packages/gcc10.rb

```diff
@@ -66,131 +100,292 @@ def self.build
+      @deletefiles.each do |f|
+        FileUtils.rm f if File.exist?(f)
+      end
+    end
+  end
+
+  def self.postinstall
+    # Only make links to unversioned in postinstall
+    gcc_version = version.split('-')[0]
+    gcc_arch = `gcc-#{gcc_version} -dumpmachine`.chomp
+    gcc_dir = "gcc/#{gcc_arch}/#{gcc_version}"
+    Dir.chdir "#{CREW_PREFIX}/bin/" do
+      Dir.glob("#{gcc_arch}-*-#{gcc_version}").each do |f|
+        @basefile_nover = f.split(/-#{gcc_version}/, 2).first
+        puts "Symlinking #{f} to #{@basefile_nover}"
+        FileUtils.ln_sf f, @basefile_nover
+        # @basefile_noarch=f.split(/#{gcc_arch}-/, 2).last
+        # puts "Symlinking #{f} to #{@basefile_noarch}"
+        # FileUtils.ln_sf f, @basefile_noarch
+        @basefile_noarch_nover = @basefile_nover.split(/#{gcc_arch}-/, 2).last
+        puts "Symlinking #{f} to #{@basefile_noarch_nover}"
+        FileUtils.ln_sf f, @basefile_noarch_nover
+        @basefile_noarch_nover_nogcc = @basefile_noarch_nover.split(/gcc-/, 2).last
+        puts "Symlinking #{f} to #{gcc_arch}-#{@basefile_noarch_nover_nogcc}"
+        FileUtils.ln_sf f, "#{gcc_arch}-#{@basefile_noarch_nover_nogcc}"
+      end
+      Dir.glob("*-#{gcc_version}").each do |f|
+        @basefile_nover = f.split(/-#{gcc_version}/, 2).first
+        puts "Symlinking #{f} to #{@basefile_nover}"
+        FileUtils.ln_sf f, @basefile_nover
+      end
+      # many packages expect this symlink
+      puts "Symlinking gcc-#{gcc_version} to cc"
+      FileUtils.ln_sf "gcc-#{gcc_version}", 'cc'
+    end
+    # make sure current version of gcc LTO plugin for Gold linker is installed.
+    FileUtils.mkdir_p "#{CREW_LIB_PREFIX}/bfd-plugins/"
+    puts "Symlinking #{CREW_PREFIX}/libexec/#{gcc_dir}/liblto_plugin.so to #{CREW_LIB_PREFIX}/bfd-plugins/"
+    FileUtils.ln_sf "#{CREW_PREFIX}/libexec/#{gcc_dir}/liblto_plugin.so", "#{CREW_LIB_PREFIX}/bfd-plugins/"
+    # binutils makes a symlink here, but just in case it isn't there.
+    if ARCH == 'x86_64'
+      FileUtils.mkdir_p "#{CREW_PREFIX}/lib/bfd-plugins/"
+      puts "Symlinking #{CREW_PREFIX}/libexec/#{gcc_dir}/liblto_plugin.so to #{CREW_PREFIX}/lib/bfd-plugins/"
+      FileUtils.ln_sf "#{CREW_PREFIX}/libexec/#{gcc_dir}/liblto_plugin.so", "#{CREW_PREFIX}/lib/bfd-plugins/"
+    end
+    puts 'If you do not need gcc you can uninstall gcc to save space with'.lightgreen
+    puts '"crew update ; crew upgrade ; crew remove gcc10 gcc11"'.lightgreen
+    # Remove all conflicting files from conflicting packages / reinstall
+    # This should be at the very end.
+    conflict_packages = %w[libssp]
+    conflict_packages.each do |package|
+       file = File.open("#{CREW_META_PATH}#{package}.filelist").read
+       file.each_line do |line|
+         FileUtils.rm "/#{line}" if File.exist?("/#{line}")
+       end
+      # Reinstall these conflicting packages since we may have overwritten them.
+      system "crew reinstall #{package}"
```

Comment:
I'm thinking the symlinks should go in the install section instead so they can be tracked?  Seems like there could be a more elegant way to do this?  Why don't we just not install the conflicting files from gcc in the first place?

Reply:
This particular code block is from when I had libgcc_s1 also pulled out to a different package. libssp should not conflict with gcc. 

In regards to the symlinks... they're there in postinstall as a method of avoiding conflicts with other gcc installs which could in theory be installed alongside this.

With the postinstall as it exists now, that is possible... at least for the executables, but not for the libraries (which are per-gcc -version and are all in the common lib folder.)

Why not also consider the alternative of appending files to the CREW_META filelist from post-install code blocks?

We need to create an array of files that are being created, and then append them to the filelist...

Reply:
You don't have to do all that if you put everything in install instead of postinstall. :)  The problem with not tracking the symlinks is they become dangling after package removal.